src/cuda_eval.cpp

```diff
@@ -920,15 +964,16 @@ static void jitc_cuda_render_scatter_kahan(const Variable *v, uint32_t v_index)
         index, index, value, ptr_1,
         index, index, value, ptr_2);
 
-    const char* op_suffix = jitc_is_single(value) ? ".ftz" : "";
+    const char* op_suffix = jitc_is_single(value) ? ".ftz"  : "";
+    const char* red_atom_suffix = jitc_is_half(value) ? ".noftz" : "";
 
     fmt("    {\n"
         "        .reg.$t %before, %after, %value, %case_1, %case_2;\n"
         "        .reg.$t %abs_before, %abs_value, %result;\n"
         "        .reg.pred %cond;\n"
         "\n"
-        "        mov.$t %value, $v;\n"
-        "        atom.global.add.$t %before, [%rd2], %value;\n"
+        "        mov.$b %value, $v;\n"
```

Comment:
If float16 atomics aren't properly supported, then I don't understand why there are changes here. I would prefer to keep the patch small in this case, unless I am misunderstanding something?